qiskit/providers/aer/backends/wrappers/CMakeLists.txt

```diff
@@ -26,11 +26,28 @@ if(APPLE)
     unset(PYTHON_LIBRARIES)
 endif()
 
+set(_find_pybind_includes_command "
```

Comment:
CMake can be tough in the beginning :(
The correct way to implement any dependency on CMake is using: `find_package()` function.
There are instructions on how to implement pybind11 with CMake [here](https://pybind11.readthedocs.io/en/stable/compiling.html#advanced-interface-library-target).
Basically, `find_package(pybind11 REQUIRED)` should be smart enough to find the installation of pybind and export all the functions and set all the variables you need to use later. Then you just need to tell CMake to link with pybind using this other function:
```
target_link_libraries(qasm_controller_wrapper
    ${AER_LIBRARIES}
    ${PYTHON_LIBRARIES}
    pybind11::embed)
```
Try implementing this way, as the doc says and ping me if you run into problems.

Reply:
Yes - so that is how we used to deal with the pybind package when we were building with conda - the unfortunate part of using pip is that the pypi package does not include the cmake modules - if you have pybind11 pip installed already you can run 'pip show -f pybind11' to see that the only files included are the c++ header files 
I really don't know why pybind/pypi decided to do it this way - the conda package is different in that its installation includes the cmake modules (although perhaps in a non-standard location)

Reply:
Oh! this very unfortunate... seems like this problem was already brought here: https://github.com/pybind/pybind11/issues/1628. 
Ok, I'd move this code into it's own `cmake/pybind11Config.cmake` file so we can use `find_package()` in config mode:
```
find_package(pybind11 CONFIG REQUIRED)
```
This will look into our  `cmake/` directory for `pybind11Config.cmake` file and export PYBIND_INCLUDE_DIRS from there. I can help you with this too.

Reply:
I moved the finding of the pybind headers to pybind11Config.cmake - I think we may want to implement some kind of exception for the shadowed-declaration warnings that are produced by pybind since they're quite excessive and probably (?) not relevant - thoughts?